src/lib.rs

```diff
@@ -45,15 +45,57 @@ pub struct Header {
     pub generator: u32,
 }
 
+/// Forces early fragment tests. See https://www.khronos.org/opengl/wiki/Early_Fragment_Test.
+/// In GLSL: layout(early_fragment_tests) in;
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub struct EarlyFragmentTest {}
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum FragmentDepthWrite {
+    ///
+    Greater,
+
+    ///
+    Less,
+
+    ///
+    Unchanged,
+}
+///
+#[derive(Clone, Copy, Debug, PartialEq)]
+pub enum PixelCenter {
```

Comment:
Can we not have that? I recall the only API using 0.5 pixel offset was DX9

Reply:
SPIR-V doesn't care. You must support them. Remember, this has nothing to do what you do on the API side of things. This is simply a **choice** that a shader writer can and will use.

One is native to HLSL and one is native to (old) GLSL. Modern GLSL allows to specify It and thus doesn't care either.

When translating HLSL -> GLSL an easy thing is to force Offset option. The other way around will be easy if Offset option was specified in the original shader. Otherwise, we must add vec2(0.5, 0.5) to any access.

Reply:
You are right that only DX9 has it. However, it is allowed in SPIR-V (no extensions, no higher versions than 1.0 needed) as well is GLSL.

I would personally leave as It is. It will be there for GLSL<->SPIR-V pathways. And if somebody wants then they can implement conversion from integer to offset for compability.

Reply:
Does, say, SPIRV-Cross handle it in any way? If it's not practical to support, we shouldn't even try.
There is no saying that Naga has to support every weird SPIR-V under the sun. We are defining some practical boundaries.